backend/static/frontend/org/riceapps/controllers/schedule_planner_controller.js

```diff
@@ -83,6 +84,25 @@ SchedulePlannerController.prototype.onCRNViewClick_ = function(event) {
 };
 
 /**
+ * Event handler; called when clear playground button is clicked. Clears all courses from the playground.
+ * @param {SchedulePlannerEvent} event
+ * @private
+ */
+SchedulePlannerController.prototype.onClearPlaygroundClick_ = function(event) {
+  if (this.userModel_ != null && event.type == SchedulePlannerEvent.Type.CLEAR_PLAYGROUND_CLICK){
+    var course_views = this.view_.getPlaygroundView().getChildren().slice();
+    for (var i = 0; i < course_views.length; i++) {
+      // Update the user model.
+      this.userModel_.removeCoursesFromPlayground([course_views[i].getCourseModel()]);
+
+      // Dispose of the view.
+      event.target.removeChild(course_views[i], true);
```

Comment:
I would use getPlaygroundView() in place of even.target since it's more readable.